controllers/configurationpolicy_controller.go

```diff
@@ -860,29 +933,40 @@ func (r *ConfigurationPolicyReconciler) handleObjects(
 	objNames := []string{}
 	remediation := policy.Spec.RemediationAction
 
-	name, kind, metaNamespace := getDetails(unstruct)
-	if metaNamespace != "" {
-		namespace = metaNamespace
-		log = log.WithValues("namespace", metaNamespace)
+	// Re-fetch details in case they were templated
+	getDetails(&objDetails, unstruct)
```

Comment:
Can you call this in `handleObjectTemplates` right after template processing instead?

Calling it here causes the implementation details of `handleObjectTemplates` to be leaked into `handleObjects`. It also doesn't need to be called for *every* namespace, just every object template.

Reply:
Good point. Yes, I'll move it.